library/std/src/sys/hermit/os.rs

```diff
@@ -112,6 +112,34 @@ pub struct Env {
     iter: vec::IntoIter<(OsString, OsString)>,
 }
 
+// FIXME(https://github.com/rust-lang/rust/issues/114583): Remove this when <OsStr as Debug>::fmt matches <str as Debug>::fmt.
+pub struct EnvStrDebug<'a> {
```

Comment:
Could we avoid the code duplication here and move `EnvStrDebug` into common code?

Reply:
Not without introducing additional allocations.

Reply:
Perhaps I should elaborate. The various `Env` structures contain iterators - even when they are backed by vectors - so it isn't possible to implement `Debug` without iterating those iterators. This means the iterators must be cloned. In the case of a vector-backed iterator, cloning would allocate, which this approach avoids.